WooCommerce/Classes/System/WooConstants.swift

```diff
@@ -270,10 +270,10 @@ private extension WooConstants.URLs {
     /// Convert a `string` to a `URL`. Crash if it is malformed.
     ///
     private static func trustedURL(_ url: String) -> URL {
-        if let url = URL(string: url) {
-            return url
-        } else {
-            fatalError("Expected URL \(url) to be a well-formed URL.")
+        guard let url = URL(string: url) else {
+            let fallback = URL(string: url.addingPercentEncoding(withAllowedCharacters: NSCharacterSet.urlQueryAllowed) ?? "")
+            return fallback ?? URL(string: "")!
         }
+        return url
```

Comment:
Super nit: Since we want to crash this method when the URL cannot be parsed, this can be updated for readability:
```suggestion
        if let url = URL(string: url) {
            return URL
        } else if let escapedString = url.addingPercentEncoding(withAllowedCharacters: NSCharacterSet.urlQueryAllowed),
            let escapedURL = URL(string: escapedString) {
                return escapedURL
        } else {
            fatalError("Expected URL \(url) to be a well-formed URL.")
        }
```

Reply:
Good point, changed here: https://github.com/woocommerce/woocommerce-ios/pull/8805/commits/a68ef434cefc58085d059155ecdff73aa2366751

As a side note I'm not positive that we should crash if URLs are un-escaped in this way from an UX perspective, or just redirect users to a different WooCommerce URL, but that's something I'll explore on a different PR 👍